_rules/SC2-1-2-no-keyboard-trap.md

```diff
@@ -0,0 +1,241 @@
+---
+name: No keyboard trap
+rule_type: composite
+description: |
+  This rule checks if it is possible to use standard keyboard navigation to navigate through all content on a web page without becoming trapped in any element, or that it is possible to use non-standard keyboard navigation to navigate through content where focus is trapped when using standard ways of keyboard navigation.
+
+success_criterion:
+- 2.1.2 # No Keyboard Trap (A)
+
+atomic_rules:
+- [No keyboard trap standard navigation](SC2-1-2-no-keyboard-trap-standard-navigation)
+- [No keyboard trap non-standard navigation](SC2-1-2-no-keyboard-trap-non-standard-navigation)
+
+authors:
+- Geir Sindre Fossøy
+- Dagfinn Rømen
+- Anne Thyme Nørregaard
+---
+
+## Aggregation Definition
+
+### Applicability
+
+The rule applies to any HTML or SVG element on a web page that is [focusable](#focusable).
+
+Note: The WCAG 2.0 success criterion 2.1.2 applies to all content, not just HTML or SVG, where focus can be moved to through keyboard navigation.
+
+### Expectation
+
+For each test target, the outcome of one of the following rules is "passed":
+
+- [No keyboard trap standard navigation](SC2-1-2-no-keyboard-trap-standard-navigation)
+- [No keyboard trap non-standard navigation](SC2-1-2-no-keyboard-trap-non-standard-navigation)
+
+## Assumptions
+
+_There are currently no assumptions._
+
+## Accessibility support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.1.2: No Keyboard Trap](https://www.w3.org/WAI/WCAG21/Understanding/no-keyboard-trap.html)
+- [G21: Ensuring that users are not trapped in content](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G21)
+
+## Test Cases
```

Comment:
Probably not the right place to ask: do we need to have test cases in composite rules? Can't we just specify the outcome based on the outcome of the atomic rules (which is what the expectation does)?

Reply:
I'd question this as well. We do end up with a lot of duplicated test cases otherwise.

Reply:
The spec says we need them. And I tend to agree. The combination of atomic rules could require completely different test cases for the composite rules. Maybe not in this case but generally they are important to have.

Reply:
It's still a combination of atomic rules. As I see it, the implementation of a composite requires running the atomic rules on the test case and combine the outcome of the atomic rules. I guess what I mean is that validation of a composite rule is validation that the logic expressed in the expectation was correctly implemented. It's not a validation of any HTML code.

Reply:
We were wondering about this as well. Maybe you should raise a new issue for this? We will keep them in for now, as they are in the current spec.

Reply:
@kasperisager see issue #297 . We will keep it in for now, until the changes in the ACT format are ready.